util/progress/progressui/json.go

```diff
@@ -0,0 +1,61 @@
+package progressui
+
+import (
+	"time"
+
+	"github.com/moby/buildkit/solver/pb"
+	digest "github.com/opencontainers/go-digest"
+)
+
+// SolveStatus holds the latest status update for a vertex by its digest.
+type SolveStatus struct {
```

Comment:
Maybe `JSONProgress` is clearer name.

Reply:
Maybe `SolveStatusUpdate`? This struct doesn't really communicate the JSON progress. It's just a struct that happens to have the metadata annotations to serialize to JSON with a certain format.

I do think it's unclear this is an update to the solve status and doesn't communicate the entirety of the process.

Reply:
Maybe `VertexProgress` as this object is scoped to a single vertex?

Reply:
I like this name. I'll make that change.